From d61207f8ff63d5c35fc441f59bc7b0b4fc359465 Mon Sep 17 00:00:00 2001
From: Speedy11CZ <michalspisak53@gmail.com>
Date: Wed, 11 Aug 2021 01:56:50 +0200
Subject: [PATCH] Fetch modules from the ArcadiaMC Jenkins


diff --git a/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java b/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
index 338c30d3..df4a36da 100644
--- a/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
+++ b/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
@@ -3,8 +3,10 @@ package net.md_5.bungee.module;
 import java.io.IOException;
 import java.net.URL;
 import java.net.URLConnection;
-import java.nio.file.Files;
 import java.nio.file.StandardCopyOption;
+
+import com.google.common.io.ByteStreams;
+import com.google.common.io.Files;
 import lombok.Data;
 import net.md_5.bungee.Util;
 
@@ -18,21 +20,14 @@ public class JenkinsModuleSource implements ModuleSource
         System.out.println( "Attempting to Jenkins download module " + module.getName() + " v" + version.getBuild() );
         try
         {
-            final String url = String.format(
-                "https://papermc.io/api/v2/projects/%1$s/versions/%2$s/builds/%3$s/downloads/%4$s-%2$s-%3$s.jar",
-                "waterfall",
-                net.md_5.bungee.api.ProxyServer.getInstance().getVersion().split(":")[2].split("-")[0],
-                version.getBuild(),
-                module.getName()
-            );
-            URL website = new URL( url );
+            URL website = new URL( "https://ci.arcadiamc.cz/job/ArcadeCord/job/main/" + version.getBuild() + "/artifact/ArcadeCord-Proxy/module/" + module.getName().replace( '_', '-' ) + "/target/" + module.getName() + ".jar" );
             URLConnection con = website.openConnection();
             // 15 second timeout at various stages
             con.setConnectTimeout( 15000 );
             con.setReadTimeout( 15000 );
             con.setRequestProperty( "User-Agent", "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36" );
-
-            Files.copy( con.getInputStream(), module.getFile().toPath(), StandardCopyOption.REPLACE_EXISTING );
+            
+            Files.write( ByteStreams.toByteArray( con.getInputStream() ), module.getFile() );
             System.out.println( "Download complete" );
         } catch ( IOException ex )
         {
-- 
2.24.1.windows.2

